File: src/pathfinder_combat_simulator/core/combat_log.py
from typing import List

class CombatLog:
    """Handles logging of combat events"""
    def __init__(self):
        self.log_entries: List[str] = [] # Added type hint

    def add_entry(self, message: str):
        """Add an entry to the combat log"""
        self.log_entries.append(message)
        print(message)  # Also print for immediate feedback

    def get_full_log(self) -> str:
        """Get the complete combat log"""
        return "\n".join(self.log_entries)

    def clear(self):
        """Clear the combat log"""
        self.log_entries.clear()

File: src/core/combatLog.ts
export class CombatLog {
  private logEntries: string[] = [];

  constructor() {}

  addEntry(message: string): void {
    this.logEntries.push(message);
    console.log(message); // Also print for immediate feedback
  }

  getFullLog(): string {
    return this.logEntries.join("\n");
  }

  getLogEntries(): string[] {
    return [...this.logEntries]; // Return a copy
  }

  clear(): void {
    this.logEntries = [];
  }
}
